images/openwisp_base/Dockerfile

```diff
@@ -7,67 +7,68 @@ FROM python:3.10-slim-bullseye AS SYSTEM
 
-RUN pip install --user --upgrade pip
-ARG OPENWISP_MONITORING_SOURCE=https://github.com/openwisp/openwisp-monitoring/tarball/docker-openwisp-usage
-RUN pip install --user --upgrade ${OPENWISP_MONITORING_SOURCE}
-ARG OPENWISP_FIRMWARE_SOURCE=https://github.com/openwisp/openwisp-firmware-upgrader/tarball/docker-openwisp-usage
-RUN pip install --user --upgrade ${OPENWISP_FIRMWARE_SOURCE}
-ARG OPENWISP_TOPOLOGY_SOURCE=https://github.com/openwisp/openwisp-network-topology/tarball/docker-openwisp-usage
-RUN pip install --user --upgrade ${OPENWISP_TOPOLOGY_SOURCE}
-ARG OPENWISP_RADIUS_SOURCE=https://github.com/openwisp/openwisp-radius/tarball/docker-openwisp-usage
-RUN pip install --user --upgrade ${OPENWISP_RADIUS_SOURCE}
+RUN pip install --user --upgrade pip~=22.0.4
+ARG OPENWISP_MONITORING_SOURCE=openwisp-monitoring
+RUN pip install --user --upgrade ${OPENWISP_MONITORING_SOURCE}~=1.0.0
+ARG OPENWISP_FIRMWARE_SOURCE=openwisp-firmware-upgrader
+RUN pip install --user --upgrade ${OPENWISP_FIRMWARE_SOURCE}~=1.0.0
+ARG OPENWISP_TOPOLOGY_SOURCE=openwisp-network-topology
+RUN pip install --user --upgrade ${OPENWISP_TOPOLOGY_SOURCE}~=1.0.0
+ARG OPENWISP_RADIUS_SOURCE=openwisp-radius
+RUN pip install --user --upgrade ${OPENWISP_RADIUS_SOURCE}~=1.0.0
 
 # here we try to install custom versions of the modules only if the
 # supplied argument does not equal the default value, because
 # otherwise these modules will have already been installed above
 ARG OPENWISP_IPAM_SOURCE=default
 RUN if [ "$OPENWISP_IPAM_SOURCE" != "default" ] ; then \
-    pip install --user --upgrade ${OPENWISP_IPAM_SOURCE}; \
+    pip install --user --upgrade ${OPENWISP_IPAM_SOURCE}~=1.0.0; \
     fi
 ARG OPENWISP_CONTROLLER_SOURCE=default
 RUN if [ "$OPENWISP_CONTROLLER_SOURCE" != "default" ] ; then \
-    pip install --user --upgrade ${OPENWISP_CONTROLLER_SOURCE}; \
+    pip install --user --upgrade ${OPENWISP_CONTROLLER_SOURCE}~=1.0.0; \
     fi
 ARG OPENWISP_NOTIFICATION_SOURCE=default
 RUN if [ "$OPENWISP_NOTIFICATION_SOURCE" != "default" ] ; then \
-    pip install --user --upgrade ${OPENWISP_NOTIFICATION_SOURCE}; \
+    pip install --user --upgrade ${OPENWISP_NOTIFICATION_SOURCE}~=1.0.0; \
     fi
 ARG OPENWISP_USERS_SOURCE=default
 RUN if [ "$OPENWISP_USERS_SOURCE" != "default" ] ; then \
-    pip install --user --upgrade --force-reinstall ${OPENWISP_USERS_SOURCE}; \
+    pip install --user --upgrade --force-reinstall ${OPENWISP_USERS_SOURCE}~=1.0.0; \
     fi
 ARG OPENWISP_UTILS_SOURCE=default
 RUN if [ "$OPENWISP_UTILS_SOURCE" != "default" ] ; then \
-    pip install --user --upgrade --force-reinstall ${OPENWISP_UTILS_SOURCE}; \
+    pip install --user --upgrade --force-reinstall ${OPENWISP_UTILS_SOURCE}~=1.0.1; \
     fi
 ARG DJANGO_X509_SOURCE=default
 RUN if [ "$DJANGO_X509_SOURCE" != "default" ]; then \
-    pip install --user --upgrade --force-reinstall ${DJANGO_X509_SOURCE}; \
+    pip install --user --upgrade --force-reinstall ${DJANGO_X509_SOURCE}~=1.1.0; \
```

Comment:
These changes here are incorrect, it will break the possibility to install custom openwisp-modules.

Can you disable the linter only in this part of the dockerfile?

Reply:
Yes :+1: